Cookbook.API/Controllers/RecipesController.cs

```diff
@@ -0,0 +1,117 @@
+﻿using Cookbook.API.Mapping;
+using Cookbook.Application.Database;
+using Cookbook.Application.Services;
+using Cookbook.Contracts.Requests;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Cookbook.API.Controllers
+{
+    [ApiController]
+    public class RecipesController : ControllerBase
+    {
+        private readonly IRecipebookReadServices recipebookReadServices;
+        private readonly IRecipebookWriteServices recipebookWriteServices;
+
+        public RecipesController(IRecipebookReadServices recipebookReadServices, 
+                                 IRecipebookWriteServices recipebookWriteServices)
+        {
+            this.recipebookReadServices = recipebookReadServices;
+            this.recipebookWriteServices = recipebookWriteServices;
+        }
+
+        [HttpPost(ApiEndPoints.Recipes.Create)]
+        public async Task<IActionResult> Create([FromBody] CreateRecipeRequest request,
+                                                CancellationToken token)
+        {
+            var recipe = request.MapToRecipe();
+
+            await recipebookWriteServices.CreateAsync(recipe, token);
+
+            var response = recipe.MapToResponse();
+
+            return CreatedAtAction(nameof(Get), new { idOrSlug = response.Id }, response);
+        }
+
+        [HttpGet(ApiEndPoints.Recipes.Get)]
+        public async Task<IActionResult> Get([FromRoute] string idOrSlug,
```

Comment:
En ezeket szetvalasztanam ket endpoint-ba, de lehet jobb igy

Reply:
a videoban egyben volt  de szet is szedhetem, viszont akkor h nezne ki v hogy valasztja szet?

[HttpGet(ApiEndPoints.Recipes.Get)]
        public async Task<IActionResult> Get([FromRoute] string slug,
                                                CancellationToken token)
        {
            var recipe =  await recipebookReadServices.GetBySlugAsync(slug, token);

            if (recipe is null)
            {
                return NotFound();
            }

            var response = recipe.MapToResponse();
            return Ok(response);
        }

[HttpGet(ApiEndPoints.Recipes.Get)]
        public async Task<IActionResult> Get([FromRoute] Guid id,
                                                CancellationToken token)
        {
            var recipe = await recipebookReadServices.GetByIdAsync(id, token);

            if (recipe is null)
            {
                return NotFound();
            }

            var response = recipe.MapToResponse();
            return Ok(response);
        }

es a kod majd eldonti h amit kapott az Guid v string? nem fogja megzavarni h a JSONban string a guid?

Reply:
Hmmmm, igen en igy valasztanam szet, de jo kerdes, lehet megzavarja es ezert volt egyben. Probald ki esetleg, azzal is tanulsz, nem tudom hogy "felismeri-e" a guid-ot vagy lehet-e valahogy "priorizalni" hogy kezd a guid-dal, ha nem guid akkor menj a string-re.